lib/jekyll/excerpt.rb

```diff
@@ -105,7 +105,9 @@ def inspect
     #
     # Returns excerpt String
     def extract_excerpt(post_content)
-      separator     = site.config['excerpt_separator']
+      separator     = !post.excerpt_separator.empty? &&
+                       post.excerpt_separator ||
+                       site.config['excerpt_separator']
```

Comment:
The alignment here seems a bit weird: since the `!` only applies to `post.excerpt_separator.empty?`, the other two probably shouldn't be indented past the `!` (in other words, maybe unindent the above 2 lines by 1 space).